### PR TITLE
Beta fixes

### PR DESCRIPTION
**Pull request by sakshikukreja14**

### CODE REVIEW

File: src/gui/mzroll/mainwindow.cpp [hunk 1]
```diff
@@ -2485,35 +2488,34 @@ int MainWindow::loadMetaCsvFile(string filename){
...
             QColor color;
             
-            if(colorName.empty()) 
-                continue;    
-            if (colorName[0] == '#') {
-                color.setNamedColor(QString::fromStdString(colorName));
-                color.getRgbF(&redF, &greenF, &blueF, &alpha);
-            } else {
-                try {
+            if(!colorName.empty()) {
```
Comment (saifulbkhan) [inline src/gui/mzroll/mainwindow.cpp:2500]: @sakshikukreja14 Any reason why the code from here on was modified in this patch? Once you've read the JSON from its correct path, I would assume there is no reason to make any further changes.

Comment (sakshikukreja14) [reply src/gui/mzroll/mainwindow.cpp:2499]: Earlier, we were using "continue" instead. But it was skipping the implementation of assigning cohort values if color name was empty.

File: src/core/libmaven/peakdetector.cpp [hunk 2]
```diff
@@ -156,15 +156,30 @@ void PeakDetector::identifyFeatures(const vector<Compound*>& identificationSet)
...
         massSlicer.generateAdductSlices(identificationSet, true, false);
+        massSlicer.generateCompoundSlices(compoundsWithNeutralMassOnly, false);
+        massSlicer.generateCompoundSlices(compoundsWithRawMzOnly, false);
     } else if (_mavenParameters->pullIsotopesFlag) {
         massSlicer.generateIsotopeSlices(identificationSet);
+        massSlicer.generateCompoundSlices(compoundsWithNeutralMassOnly, false);
+        massSlicer.generateCompoundSlices(compoundsWithRawMzOnly, false);
     } else if (_mavenParameters->searchAdducts) {
         massSlicer.generateAdductSlices(identificationSet);
+        massSlicer.generateCompoundSlices(compoundsWithRawMzOnly, false);
```
Comment (sakshikukreja14) [inline src/core/libmaven/peakdetector.cpp:182]: Why not compounds with neutral mass added here?

File: src/core/libmaven/peakdetector.cpp [hunk 3]
```diff
@@ -992,6 +992,7 @@ void PeakDetector::linkParentIsotopeRange(PeakGroup& parentGroup,
                 child->deletePeak(sample);
         }
 
+        delete_all(eics);
```
Comment (sakshikukreja14) [inline src/core/libmaven/peakdetector.cpp:995]: Was this the reason for crash while detecting peaks, reported by testing team?

File: src/core/libmaven/peakdetector.cpp (hunk 2 above)
Comment (saifulbkhan) [reply src/core/libmaven/peakdetector.cpp:182]: Right, that was intentional. For isotopes, at least formula is required, but for adducts, either formula or neutral mass is sufficient for calculating adduct mass. So compounds that have formula or neutral mass do not need any special treatment here.

File: src/core/libmaven/peakdetector.cpp (hunk 3 above)
Comment (saifulbkhan) [reply src/core/libmaven/peakdetector.cpp:995]: Yes. Entire EICs for each peak-group (and its children) were being created but not being deallocated during this step. And each EIC is a large object - this was a very bad memory leak.

**Review by saifulbkhan — APPROVED** [review]

**Pull request merged**